### PR TITLE
[wasm][tests] Use a polyfill library to provide Web Crypto API

### LINKED ISSUE #42730: [wasm][tests] Use a polyfill for WebCrypto API support
Right now the System.Security.Cryptography tests uses a generic [`getRandomValues`](https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues).

Microsoft provides a [Web Cryptography](https://github.com/microsoft/MSR-JavaScript-Crypto) polyfill API that can be used to provide the interface for testing within the CI.

Without a [`getRandomValues`](https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues) implementation the following is thrown in the CI.

```
ABORT: no cryptographic support found for random_device. consider polyfilling it if you want to use something insecure like Math.random(), e.g. put this in a --pre-js: var crypto = { getRandomValues: function(array) { for (var i = 0; i < array.length; i++) array[i] = (Math.random()*256)|0 } };

```

### PR DESCRIPTION
**Pull request by kjpou1**
Use Microsoft's provided [Web Cryptography](https://github.com/microsoft/MSR-JavaScript-Crypto) polyfill API in the test CI.

- Add a new parameter to the `runtime.js` to load extra scripts such as this one:  `--extra-scripts=`
- The `--extra-scripts=` parameter takes comma separated list of scripts to load.
- An actual implementation of using the Web Crypto API [`getRandomValues`](https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues) can be found in the PR https://github.com/dotnet/runtime/pull/42728

close https://github.com/dotnet/runtime/issues/42730

part of https://github.com/dotnet/runtime/issues/40074

### CODE REVIEW

**Comment by steveisok** [comment]
Ah, cool.  We'll have something more legit when running against v8.

**Review by steveisok — DISMISSED** [review]

**Comment by marek-safar** [comment]
Why do we need this? This looks like the scenario we don't need to support and the one we need to support cannot be tested this way

File: src/mono/wasm/polyfills/msrcrypto.js [hunk 1]
```diff
@@ -0,0 +1,9404 @@
+//*******************************************************************************
+//
+//    Copyright 2020 Microsoft
```
Comment (jkotas) [inline src/mono/wasm/polyfills/msrcrypto.js:3]: This should get mention in https://github.com/dotnet/runtime/blob/master/THIRD-PARTY-NOTICES.TXT

Comment (kjpou1) [reply src/mono/wasm/polyfills/msrcrypto.js:3]: addressed in recent commits

**Comment by kjpou1** [comment]
It provides the same interface as the browser web cryptography api that can be run in the CI test harness without the need for  browser frontend to provide them.

**Comment by steveisok** [comment]
> Why do we need this? This looks like the scenario we don't need to support and the one we need to support cannot be tested this way

@marek-safar While not absolutely necessary, I think it's more 'complete' to have the polyfill.  

```
var crypto = {	
		getRandomValues: function (buffer) {	
			buffer[0] = (Math.random()*256)|0;	
		}	
	}
```

**Review by marek-safar — CHANGES_REQUESTED** [review]
@steveisok the idea was to cut any custom implementation for crypto by calling into browser's APIs. Replacing one external implementation with another proxy implementation does not help us as we'll responsible for all the deployment and security problems.

**Comment by kjpou1** [comment]
This will not be deployed with our implementations.  It is only to facilitate the testing of the API via the CI.  The test code will be written to the browser spec and the polyfill will only facilitate the test api.

**Comment by GrabYourPitchforks** [comment]
This adds ~400KB to the source repos. Since we'll probably need to keep these new dependencies up-to-date, I imagine this cost will grow over time.

Is there any other way to pull this in? For example, via a _runtime-assets_ package?

**Comment by GrabYourPitchforks** [comment]
Also, if we're going to take a dependency on the MSR polyfill, what commit hash are we basing these files from? I can send an update for Component Governance once you're finished.

**Comment by steveisok** [comment]
I don’t think it’s worth taking this dependency.  I would recommend we close the PR.

**Comment by kjpou1** [comment]
Closing the PR as it brings in an unwanted dependency as per the comments above.  Thanks everyone for providing feedback.

**Pull request closed without merging**